extensions/NOname-awa/global-coordinate.js

```diff
@@ -0,0 +1,484 @@
+				args.y = 0;
+			}
+			if (isNaN(args.r)) {
+				args.r = 0;
+			}
+			if (isNaN(args.s)) {
+				args.s = 0;
+			}
+			target.setSize((args.s / 100) * gs[args.screen - 1]);
+			target.setDirection((- ((args.r - (gr[args.screen - 1] - 90) - 90))) + 90);
+			if (rm[args.screen - 1] == 1) {
+				target.setXY(
+					(gx[args.screen - 1] / 100 * gs[args.screen - 1]) + (gs[args.screen - 1] / 100) * ((- (args.x)) * Math.cos((((- ((gr[args.screen - 1] - 90 - 90))) + 90)) / 180 * Math.PI) - (- (args.y)) * Math.sin((((- ((gr[args.screen - 1] - 90 - 90))) + 90)) / 180 * Math.PI)),
+					(gy[args.screen - 1] / 100 * gs[args.screen - 1]) + (gs[args.screen - 1] / 100) * ((- (args.x)) * Math.sin((((- ((gr[args.screen - 1] - 90 - 90))) + 90)) / 180 * Math.PI) + (- (args.y)) * Math.cos((((- ((gr[args.screen - 1] - 90 - 90))) + 90)) / 180 * Math.PI)),
+					true
+				);
+			}
+			else {
+				target.setXY(
+					(gs[args.screen - 1] / 100) * ((- (args.x + gx[args.screen - 1])) * Math.cos((((- ((gr[args.screen - 1] - 90 - 90))) + 90)) / 180 * Math.PI) - (- (args.y + gy[args.screen - 1])) * Math.sin((((- ((gr[args.screen - 1] - 90 - 90))) + 90)) / 180 * Math.PI)),
+					(gs[args.screen - 1] / 100) * ((- (args.x + gx[args.screen - 1])) * Math.sin((((- ((gr[args.screen - 1] - 90 - 90))) + 90)) / 180 * Math.PI) + (- (args.y + gy[args.screen - 1])) * Math.cos((((- ((gr[args.screen - 1] - 90 - 90))) + 90)) / 180 * Math.PI)),
+					true
+				);
+			}
+		}
+		rotation_mode(args) {
+			rm[args.screen - 1] = args.m;
+		}
+		set({ x, y, r, s, screen }) {
+			if (!isNaN(x)) {
+				gx[screen - 1] = x;
+			}
+			else {
+				gx[screen - 1] = 0;
+			}
+			if (!isNaN(y)) {
+				gy[screen - 1] = y;
+			}
+			else {
+				gy[screen - 1] = 0;
+			}
+			if (!isNaN(r)) {
+				gr[screen - 1] = r;
+			}
+			else {
+				gr[screen - 1] = 0;
+			}
+			if (!isNaN(s)) {
+				gs[screen - 1] = s;
+			}
+			else {
+				gs[screen - 1] = 0;
+			}
+		}
+		Set_Co({ x, y, screen }) {
+			if (!isNaN(x + y)) {
+				gx[screen - 1] = x;
+				gy[screen - 1] = y;
+			}
+			else {
+				if (isNaN(x)) {
+					x = 0;
+				}
+				if (isNaN(y)) {
+					y = 0;
+				}
+			}
+		}
+		Set_GX({ x, screen }) {
+			if (!isNaN(x)) {
+				gx[screen - 1] = x;
+			}
+			else {
+				gx[screen - 1] = 0;
+			}
+		}
+		Set_GY({ y, screen }) {
+			if (!isNaN(y)) {
+				gy[screen - 1] = y;
+			}
+			else {
+				gy[screen - 1] = 0;
+			}
+		}
+		CX({ x, screen }) {
+			if (!isNaN(x)) {
+				gx[screen - 1] += x;
+			}
+		}
+		CY({ y, screen }) {
+			if (!isNaN(y)) {
+				gy[screen - 1] += y;
+			}
+		}
+		Set_GR({ r, screen }) {
+			if (!isNaN(r)) {
+				gr[screen - 1] = r;
+			}
+			else {
+				gr[screen - 1] = 0;
+			}
+		}
+		TR({ r, screen }) {
+			if (!isNaN(r)) {
+				gr[screen - 1] += r;
+			}
+		}
+		TL({ r, screen }) {
+			if (!isNaN(r)) {
+				gr[screen - 1] -= r;
+			}
+		}
+		Set_si({ s, screen }) {
+			if (!isNaN(s)) {
+				gs[screen - 1] = s;
+			}
+			else {
+				gs[screen - 1] = 0;
+			}
+			if (gs < 0) {
```

Comment:
This looks wrong
```
Operator '<' cannot be applied to types 'number[]' and 'number'.ts(2365)
```

gs is an array; why is it being compared to a number? are you sure this does what you want?